deep-learning/_posts/2018-01-22-MatchingNet.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: review
+title:  "Matching Networks for One Shot Learning"
+tags:   "Deep Learning"
+author: Faezeh Amjadi
+pdf:    https://arxiv.org/pdf/1606.04080.pdf
+cite:
+  authors: "Oriol Vinyals, Charles Blundell, Timothy Lillicrap, Koray Kavukcuoglu, Daan Wierstra"
+  title:   "Matching Networks for One Shot Learning"
+  venue:   "--"
+---
+
+### Description
+
+This paper proposed a solution for small data sets which suffers from overfitting and generalizing. The authors employ a metric learning which help the network to learn rapid from small data set. They applied this idea on a small set of Image Net and they get a accuracy from 87.6% to 93.2%.
+
+
+They use the advantages of one shot learning, but for prediction on the target, they investigate a metric to found how much a unseen sample is simmilar to seen sample in a episod and this similarity can present as a weight on classes.
+This metric plays such as attention or a kernel function. The authors used a cosin distance on two vectors as a metric.
+Suppose $$(x_{i},y_{i})$$ is a given (input, class) and $$\bar{x}$$ is given input and predicted label $$\bar{y}$$ calculated as:
+
+
+$$\bar{y} = \sum_{i=1}^k a(\bar{x},x_i) y_{i} $$
+ 
+
+
+a small support set contains of $$k$$ labell example with input $$x$$ and lable $$y$$, $$ S= {(x_{i},y_{i})}_{i=1}^k$$, given a new example $$\bar{x}$$, we want to know the probability that it is an example of a given class $$ P(\bar{y}\mid\bar{x},S)$$ where $$P$$ is parametrized by a neural network. Therefore $$S$$ maps to a classifier $$ c_{s}(\bar{x})$$. Simply, suppose a task $$T$$ has a distribution over lable sets $$L$$. First a set $$L$$ from task $$T$$ is sampled, Then from these sample labels, the support set $$S$$ and a batch $$B$$ are sampled. The Maching network has to minimize the prediction of lables in batch $$B$$ conditioned on support set $$(S)$$.
+
+<img src="/deep-learning/images/MNet/1.png" width="500">
+
+
+For a similarity metric, two embeding functions $$f$$ and $$g$$ need to take similarity on feature space X. function $$g$$ has embeded $$x_{i}$$ independently from other elements but $$S$$ could be able to effect how be embeded $$\bar{x}$$ through function $$f$$. Then an attention kernel calculate cosin distance on these functions (similar to nearest neighbor).
```

Comment:
Do not forget to always start a sentence with a capital letter.  Here "Function $$g$$..."

Reply:
Not sure I understand "function $$g$$ has embeded $$x_{i}$$ independently from other elements but $$S$$ could be able to effect how be embeded $$\bar{x}$$ through function $$f$$."  Please reformulate.